velox/exec/HashTable.h

```diff
@@ -310,7 +310,19 @@ class BaseHashTable {
   virtual void prepareJoinTable(
       std::vector<std::unique_ptr<BaseHashTable>> tables,
       int8_t spillInputStartPartitionBit,
-      folly::Executor* executor = nullptr) = 0;
+      folly::Executor* executor = nullptr,
+      bool dropDuplicates = false) = 0;
+
+  /// The hash table used for join build in left semi and anti join does not
+  /// retain duplicate join keys by default. This is achieved by constructing
+  /// the hash table in the addInput phase to eliminate duplicate join keys.
+  /// When the percentage of duplicate data is small, it will adaptively adjust
+  /// to not build the hash table in the addInput phase. Instead, it operates
+  /// like other join types by reading all the data before building the hash
+  /// table. This function is used to indicate that the join hash table will not
+  /// be built during the addInput phase, and the input data will also not be
+  /// deduplicated.
```

Comment:
Can we also indicate this does not impact the containing row container in any means?